### PR TITLE
Cleanup?: Use gotos in z_play for large else blocks

### PR DESCRIPTION
**Pull request by Dragorn421**
*(comment applies to initial commit 8311212e729ed9132e940243b28a149f217927c8, I have since reduced the extent of the changes, and force-pushed to make things less confusing)*

This has the advantage of removing two indentation levels on large chunks of code in `Play_Update` and `Play_Draw`.
Also IMHO it's actually easier to read with gotos, since the typical code path does not follow gotos.
I'm also pretty sure the original code did have gotos, not that I think that has as much value as the previous points. One required goto present in master is I think a huge hint of this.

I actually think every `R_HREG_MODE` check uses a goto to skip the relevant block of code, but didn't do so in the PR since it's okay-ish without doing that and I don't think it'd be super popular.

Note the diff looks very bad. I suggest just looking at the files directly (look for `goto` usage)
Also note I moved the `roomDrawFlags` temp to another scope

This is very open to discussion on if we should (not) do less/more of this.
Note I don't think this is applicable anywhere else besides `Play_Update` and `Play_Draw`, so it's I think not opening Pandora's box to consider this.

----------

Them: I like bad boys
Devs: I use gotos in C where I could have easily done differently
(e.g. split Play_Update and Play_Draw into smaller functions :pensive:)

### CODE REVIEW

**Comment by fig02** [comment]
I cant say im the hugest fan, but I understand the reasoning for doing so.
I think the checks for things that skip large amounts of update and draw (frame advance and hreg skip draw) are better off as they were, to make it more obvious how much does get skipped. I never had a problem with the indentation it adds personally and always found that it makes sense.

The other cases with transition tile and prerender are fine with the change though imo

**Review by fig02 — APPROVED** [review]

File: src/code/z_play.c [hunk 1]
```diff
@@ -1119,168 +1119,170 @@ void Play_Draw(PlayState* this) {
...
+            goto Play_Draw_skip;
+        }
+
+    Play_Draw_DrawOverlayElements:
+        if ((R_HREG_MODE != HREG_MODE_PLAY) || R_PLAY_DRAW_OVERLAY_ELEMENTS) {
+            Play_DrawOverlayElements(this);
         }
     }
 
+Play_Draw_skip:
```
Comment (hensldm) [inline src/code/z_play.c:1284]:
```suggestion
Play_Draw_Skip:
```
Not sure we have a particular style for labels, but the other one is capitalized, so seems appropriate for this one to be as well.

**Review by hensldm — APPROVED** [review]
I am also fine with this. Helps reduce the indentations, and isn't too hard to follow.

File: src/code/z_play.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_play.c:1284]: Worth noting this extra `Play_Draw_Skip` goto introduced here is actually required in MM's `Play_DrawMain` to fix regalloc. I wonder if it would also be required in OoT's retail where these go-to signals (not sure what to call it, usually see them as `dummy:;` in MM) does affect regalloc more.

**Review by engineer124 — APPROVED** [review]
Also agree with the minor suggestion above

**Review by Roman971 — APPROVED** [review]
Not a huge fan of the gotos when we can avoid them, but I'm fine with it if others are as well

File: src/code/z_play.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_play.c:1284]: I might have a macro or two off for retail, but it does look this this goto will very likely be required for OoT retail, at least looking at 1.0's `Play_Draw`

https://decomp.me/scratch/bCuGH
